vault/vault.go

```diff
@@ -27,24 +27,14 @@ func FormatKeyForDisplay(k string) string {
 
 // Mfa contains options for an MFA device
 type Mfa struct {
-	MfaToken        string
-	MfaPromptMethod string
-	MfaSerial       string
+	TokenProvider mfa.TokenProvider
+	Serial        string
 }
```

Comment:
this struct may now be unnecessary

Reply:
I had tried unsuccessfully to get rid of that, had another go and it's gone but I'm not entirely happy with the result.